codalab/model/bundle_model.py

```diff
@@ -634,19 +634,26 @@ def add_join(table: Table, condition: Any, left_outer_join: bool = False):
 
         if user_id != self.root_user_id:
             # Restrict to the bundles that we have access to.
+            aliased_group_bundle_permission = aliased(cl_group_bundle_permission)
             add_join(
-                cl_group_bundle_permission,
-                cl_bundle.c.uuid == cl_group_bundle_permission.c.object_uuid,
+                aliased_group_bundle_permission,
+                cl_bundle.c.uuid == aliased_group_bundle_permission.c.object_uuid,
+                left_outer_join=True,
+            )
+            aliased_cl_user_group = aliased(cl_user_group)
+            add_join(
+                aliased_cl_user_group,
+                aliased_cl_user_group.c.user_id == user_id,
+                left_outer_join=True,
```

Comment:
We use aliased here to account for the case when we've already done a join on the group_bundle_permission table or the user_group table. (If you do that twice without aliasing one of them, the query is invalid according to MySQL syntax rules.)

Reply:
Does `aliased` automatically pick a unique name, or do you need to explicitly pick a name?

In a different PR, we should probably add validation [when we do the joins](https://github.com/codalab/codalab-worksheets/blob/b0d285757036ebffc81c10a6dce245fc6759c21f/codalab/model/bundle_model.py#L656-L659) to make sure that we're not joining onto the same table twice.